Index: linux-3.10.18/arch/mips/bcm47xx/board.c
===================================================================
--- linux-3.10.18.orig/arch/mips/bcm47xx/board.c
+++ linux-3.10.18/arch/mips/bcm47xx/board.c
@@ -69,7 +69,12 @@ struct bcm47xx_board_type_list1 bcm47xx_
 	{{BCM47XX_BOARD_ASUS_WL500W, "Asus WL500W"}, "WL500gW-"},
 	{{BCM47XX_BOARD_ASUS_WL520GC, "Asus WL520GC"}, "WL520GC-"},
 	{{BCM47XX_BOARD_ASUS_WL520GU, "Asus WL520GU"}, "WL520GU-"},
+	{{BCM47XX_BOARD_BELKIN_F7D3301, "Belkin F7D3301"}, "F7D3301"},
+	{{BCM47XX_BOARD_BELKIN_F7D3302, "Belkin F7D3302"}, "F7D3302"},
+	{{BCM47XX_BOARD_BELKIN_F7D3302, "Belkin F7D7302"}, "F7D7302 v1"},
 	{{BCM47XX_BOARD_BELKIN_F7D4301, "Belkin F7D4301"}, "F7D4301"},
+	{{BCM47XX_BOARD_BELKIN_F7D4302, "Belkin F7D4302"}, "F7D4302"},
+	{{BCM47XX_BOARD_BELKIN_F7D4401, "Belkin F7D4401"}, "F7D4401"},
 	{ {0}, 0},
 };
 
@@ -180,6 +184,7 @@ struct bcm47xx_board_type_list3 bcm47xx_
 	{{BCM47XX_BOARD_ZTE_H218N, "ZTE H218N"}, "0x053d", "1234", "0x1305"},
 	{{BCM47XX_BOARD_NETGEAR_WNR3500L, "Netgear WNR3500L"}, "0x04CF", "3500", "02"},
 	{{BCM47XX_BOARD_LINKSYS_WRT54GSV1, "Linksys WRT54GS V1"}, "0x0101", "42", "0x10"},
+	{{BCM47XX_BOARD_BELKIN_F7DXXXX, "Belkin F7Dxxxx"}, "0xa4cf", NULL, "0x1102"},
 	{ {0}, 0},
 };
 
@@ -270,9 +275,9 @@ static __init const struct bcm47xx_board
 	    bcm47xx_nvram_getenv("boardnum", buf2, sizeof(buf2)) >= 0 &&
 	    bcm47xx_nvram_getenv("boardrev", buf3, sizeof(buf3)) >= 0) {
 		for (e3 = bcm47xx_board_list_board; e3->value1; e3++) {
-			if (!strcmp(buf1, e3->value1) &&
-			    !strcmp(buf2, e3->value2) &&
-			    !strcmp(buf3, e3->value3))
+			if ((!e3->value1 || !strcmp(buf1, e3->value1)) &&
+			    (!e3->value2 || !strcmp(buf2, e3->value2)) &&
+			    (!e3->value3 || !strcmp(buf3, e3->value3)))
 				return &e3->board;
 		}
 	}
Index: linux-3.10.18/arch/mips/include/asm/mach-bcm47xx/bcm47xx_board.h
===================================================================
--- linux-3.10.18.orig/arch/mips/include/asm/mach-bcm47xx/bcm47xx_board.h
+++ linux-3.10.18/arch/mips/include/asm/mach-bcm47xx/bcm47xx_board.h
@@ -27,7 +27,12 @@ enum bcm47xx_board {
 	BCM47XX_BOARD_ASUS_WL700GE,
 	BCM47XX_BOARD_ASUS_WLHDD,
 
+	BCM47XX_BOARD_BELKIN_F7DXXXX,
+	BCM47XX_BOARD_BELKIN_F7D3301,
+	BCM47XX_BOARD_BELKIN_F7D3302,
 	BCM47XX_BOARD_BELKIN_F7D4301,
+	BCM47XX_BOARD_BELKIN_F7D4302,
+	BCM47XX_BOARD_BELKIN_F7D4401,
 
 	BCM47XX_BOARD_BUFFALO_WBR2_G54,
 	BCM47XX_BOARD_BUFFALO_WHR2_A54G54,
